doc/sfp-cmis/cmis-init.md

```diff
@@ -0,0 +1,447 @@
+    | UNKNOWN   | Unknown state   | N/A                   |
+    | INSERTED  | Module is newly inserted | DP_DEINIT    |
+    | DP_DEINIT | DatPath is de-initialized with tx-power turned off | AP_CONFIGURED |
+    | AP_CONFIGURED | Application configured | DP_INIT    |
+    | DP_INIT   | DatPath is initialized with tx-power turned off | DP_TXON |
+    | DP_TXON   | DatPath is initialized with tx-power turned on | READY |
+    | READY     | Transceiver is ready in the new application mode | N/A |
+    | REMOVED   | Module is removed  | N/A                |
+    | FAILED    | Initialization failed | N/A             |
+
+  - At each loop iteration of CmisManagerTask.task_worker(), the state will only be advanced by 1 and only 1 state upon success.
+  - Prior to advancing the state, CmisManagerTask should always double-check the hardware module and datapath states.  
+    - Prior to handling the CMIS state transitions, the following checkers are always performed  
+      - **Check for the transceiver presence** via sfp.get_presence(),
+        abort the initialization sequence if it's no loner present
+      - **Validate the transceiver module type** via sfp.get_transceiver_info()['type_abbrv_name'],
+        abort the initialization sequence if it's not a QSFP-DD
+      - **Validate the transceiver memory type** via sfp.get_transceiver_info()['memory_type'],
+        abort the initialization sequence if the **Paged** memory is not available.
+    - From **INSERTED** to **DP_DEINIT**  
+      Skip the initialization sequence by having the state transitioned to **READY**
+      if no application code updates and DataPath state is 4 (i.e. DataPathActivated) and
+      config state is 1 (i.e. ConfigSuccess), otherwise invoke sfp.set_cmis_application_stop()
+      and have the state transitioned to **DP_DEINIT**
+    - From **DP_DEINIT** to **AP_CONFIGURED**  
+      Stay at **DP_DEINIT** state if module state != **ModuleReady**, otherwise invoke
+      sfp.set_cmis_application_apsel() and have the state transitioned to **AP_CONFIGURED**
+    - From **AP_CONFIGURED** to **DP_INIT**  
+      Stay at **AP_CONFIGURED** state if config state != 1 (i.e. ConfigSuccess), otherwise
+      invoke sfp.set_cmis_application_start() and have the state transitioned to **DP_INIT**
+    - From **DP_INIT** to **DP_TXON**  
+      Stay at **DP_INIT** state if DataPath state != 7 (i.e. DataPathInitialized), otherwise
+      invoke sfp.set_cmis_application_txon() and have the state transitioned to **DP_TXON**
+    - From **DP_TXON** to **READY**  
+      Stay at **DP_TXON** state if DataPath state != 4 (i.e. DataPathActivated), otherwise have the
+      state transitioned to **READY**
+  - The CMIS state transition diagram  
+  ![](images/001.png)
+
+
+## sonic-platform-common/sonic_platform_base/sfp_base.py (modified)
+
+Add the following macro constants to differentiate the port/cage type from the media type fetched from the transceiver
```

Comment:
CMIS support on a module is determined from the Identifier at offset 0. For eg a new QSFP+ module can also follow CMIS. See SFF-8024 v 4.9

![image](https://user-images.githubusercontent.com/45705344/143798903-b9d04254-480a-456f-ad8c-e699d6c97687.png)

We should not decide application initialization based upon cage type

Reply:
The cage type is nothing but a simple checker to determine if the CmisManagerTask should be activated, it's not directly involved in the CmisManagerTask, honestly it's the responsibility of sonic-xcvr (sfp-refactor) to differentiate the XCVR objects upon QSFP28/QSFPDD module insetion/removals on the QSFPDD ports. 
Which is known to be broken as of now

Reply:
In other words, the SfpOptoeBase.set_cmis_application_xxx() only works when a xcvr object is created based on the media type fetched from the EEPROM, and it has nothing to do with the sfp.get_port_type().
The sfp.get_port_type() is nothing but a sanity check to make sure if the CmisManagerTask is applicable to the DUT

Reply:
e.g. CmisManagerTask will be deactivated on a DUT with SFP and QSFP ports only, since it will fail the cage/port type checker.